project_task_code/models/project_task.py

```diff
@@ -0,0 +1,51 @@
+# Copyright 2016 Tecnativa <vicent.cubells@tecnativa.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+
+class ProjectTask(models.Model):
+    _inherit = "project.task"
+
+    code = fields.Char(
+        string="Task Number",
+        required=True,
+        default="/",
+        readonly=True,
+        copy=False,
+    )
+
+    @api.constrains("code")
+    def _check_project_task_unique_code(self):
+        project_task_unique_code = (
```

Comment:
No need to check for parameter. This should pass even if the parameter is not set. Only check it if task has code.

Reply:
I think there is a confusion here in the use of the parameter : 

- the task code is generated when the module is installed
- the parameter adds the check for unicity

Reply:
Changed because now the code is only set if the sequence is set in configuration, so now the constrains check if the task has code and if any other task in the company has the same code.